core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/UIAddressBar.java

```diff
@@ -158,7 +158,10 @@ public void execute(Event<UIAddressBar> event) throws Exception {
           uiExplorer.setIsViewTag(false) ;
           uiExplorer.setSelectNode(uiExplorer.getCurrentStatePath()) ;
         } else {
-          String previousNodePath = uiExplorer.rewind() ;
+          // Retrieve previous node path by refactoring current path and adding the RootPath
+          String CurrentPath =  Text.escapeIllegalJcrChars(uiAddressBar.getUIStringInput(FIELD_ADDRESS).getValue());
+          String RootPath = uiExplorer.getRootPath();

```

Comment:
You are using an object from the session here. Let's do a test, could you open two Documents applications in two pages for two different spaces , or one for Space Documents app and the other for the Personal documents app.
`String RootPath = uiExplorer.getRootPath();`

Reply:
@ahamdi , actually in case we open 2 documents from 2 different spaces in 2 different tabs , the problem is not reproduced . i tested it several times.

Reply:
and even when we open a document from personal documents app